src/BraketSimulator.jl

```diff
@@ -381,6 +380,68 @@ function simulate(simulator::AbstractSimulator,
     return results
 end
 
+# these functions are for calls from an "external" language
+# like Python or Rust, when we're calling this package from a
+# separate process and thus don't want to have to IPC large
+# blobs of data back and forth/deal with having to serialize
+# Julia objects
+function create_sim(simulator_id::String, shots::Int)
+    return if simulator_id == "braket_sv_v2"
+        StateVectorSimulator(0, shots)
+    elseif simulator_id == "braket_dm_v2"
+        DensityMatrixSimulator(0, shots)
+    end
+end
+
+function _mmap_large_result_values(results)
+    to_mmap     = findall(rt->sizeof(rt.value) > 2^20, results.resultTypes)
+    isempty(to_mmap) && return nothing, nothing
+    mmap_files  = String[]
+    obj_lengths = Int[]
+    for r_ix in to_mmap
+        push!(obj_lengths, length(results.resultTypes[r_ix].value))
+        tmp_path, io = mktemp()
+        write(io, results.resultTypes[r_ix].value)
+        empty!(results.resultTypes[r_ix].value)
+        close(io)
+        push!(mmap_files, tmp_path)
+    end
```

Comment:
For my own understanding: you're using the variable name `mmap_files`, but actually these are just temp files on disk, is that correct? Or is there something else happening here?

Reply:
`mmap_files` is the path to each file where the big array is `mmap`-ed. These files are in `/tmp` and are deleted when the Julia process exits.